DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -93,7 +93,7 @@ internal class DataUploadWorker: DataUploadWorkerType {
                     }
                 } catch let error {
                     // If upload can't be initiated do not retry, so drop the batch:
-                    self.fileReader.markBatchAsRead(batch)
+                    self.fileReader.markBatchAsRead(batch, reason: .invalid)
```

Comment:
question

is it worth knowing the error in case of an invalid removal reason?

Reply:
Totally, and we know it from the next line that sends error telemetry:
```swift
DD.telemetry.error("Failed to initiate '\(self.featureName)' data upload", error: error)
```
I don't think makes sense to send this error as part of the metric, because the cardinality could be very high. More practical scenario that I see is: notice huge volume of "deleted because of `.invalid`" metric and then find the actual issue by analysing "Failed to initiate with `error`" logs. WDYT?

Reply:
right, we should be able to query 

```
.invalid && message.startsWith("Failed to instantiate" 
```